src/main/webapp/dev/js/main/instructorFeedbackEdit.es6

```diff
@@ -389,6 +390,74 @@ function backupQuestion(questionNum) {
                                        || $(`#questionTable-${questionNum} > .panel-body`).html();
 }
 
+function isQuestionHavingResponses(questionNum) {
+    const EDIT_STATUS = $(`#form_editquestion-${questionNum}`).attr('editstatus');
+
+    return EDIT_STATUS === 'hasResponses' || EDIT_STATUS === 'mustDeleteResponses';
+}
+
+function getDestructiveFields(questionNum) {
+    const $fields = $(`#form_editquestion-${questionNum}`)
+                     .find(':input:enabled')
+                     .not('button')
+                     .not('.nonDestructive')
+                     .not('input[name^="questiondescription"]')
+                     .not('.visibilityCheckbox');
+
+    const fieldsDict = {};
+
+    for (let i = 0; i < $fields.length; i += 1) {
+        // .get(i) returns a JS object
+        const element = $fields.get(i);
+
+        if (element.type === 'radio') {
+            fieldsDict[element.name] = $(`#form_editquestion-${questionNum} input[name="${element.name}"]:checked`).val();
+        } else if (element.type === 'checkbox') {
+            fieldsDict[element.name] = $(element).is(':checked');
+        } else {
+            fieldsDict[element.name] = element.value;
+        }
+    }
+
+    return fieldsDict;
+}
+
+/**
+ * Checks if any changes were made to the destructive fields.
+ * Returns true if changes were made, else false.
+ * @param questionNum
+ */
+function isDestructiveFieldsModifed(questionNum) {
+    const currFieldsDict = getDestructiveFields(questionNum);
+    const prevFieldsDict = destructiveFieldsBackup[questionNum];
+
+    return JSON.stringify(currFieldsDict) !== JSON.stringify(prevFieldsDict);
+}
+
+function correctEditStatusIfRequired($form) {
+    const questionNum = extractQuestionNumFromEditFormId($form.attr('id'));
+
+    if (!isQuestionHavingResponses(questionNum)) {
+        return;
+    }
+
+    if (!isDestructiveFieldsModifed(questionNum)) {
+        // Destructive changes are same before and after.
+        // The changes cancelled out each other.
+        $(`#form_editquestion-${questionNum}`).attr('editstatus', 'hasResponses');
+    } else {
+        $(`#form_editquestion-${questionNum}`).attr('editstatus', 'mustDeleteResponses');
```

Comment:
Since the mechanism does not make use of the `mustDeleteReponses` anymore, what do you think of removing it from the mechanism?

Reply:
If the `else` wasn't there, then in [this situation](https://github.com/TEAMMATES/teammates/pull/7759#issuecomment-324897305), the warning wouldn't show up because there are no JS event handlers which change `editstatus` to `mustDeleteResponses`. Due to the absence of JS event handlers `editstatus` would remain `hasResponses` and upon saving, the responses get deleted without a warning. In such a situation, the `else` block here, will add `mustDeleteResponses` if there's a destructive change, leading to a warning modal to show up. So this is required.

Reply:
Sorry, I kind of misunderstood how this method is currently used, but my understanding was that if this method return true/false, having the `mustDeleteResponses` status is not necessary. Another point was that having code setting `mustDeleteResponses` to `hasResponses` when resetting, turning into `mustDeleteResponses` when changes are applied to destructive fields are redundant. I thought it would be good to clean those parts up.

Reply:
Sorry @leeyimin , but I didn't understand what you are trying to say :sweat_smile:

Reply:
I felt that other parts of code dealing with `mustDeleteResponses` could be removed, eg in line 1165
```
$('form[id|=form_editquestion]').find(':input').not('.nonDestructive').not('.visibilityCheckbox')
            .change(...);
```
since either way before saving, a check against the backup will be preformed.

It seems like it is even possible to remove the use of `mustDeleteResponses` completely if this method returns a boolean value, which can then be used to decide if a warning modal should be shown.